src/lib/y2storage/widgets/actions_summary.rb

```diff
@@ -0,0 +1,127 @@
+# encoding: utf-8
+
+# Copyright (c) [2017] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact Novell, Inc.
+#
+# To contact Novell about this file by physical or electronic mail, you may
+# find current contact information at www.novell.com.
+
+require "yast"
+require "y2storage"
+
+Yast.import "UI"
+Yast.import "HTML"
+
+module Y2Storage
+  module Widgets
+    # Widget to draw the actions of an actiongraph
+    class ActionsSummary
+      include Yast::UIShortcuts
+      include Yast::I18n
+
+      attr_reader :id
+
+      def initialize(id, actiongraph)
+        @id = id
+        @actiongraph = actiongraph
+        @collapsed_subvolumes = true
+      end
+
+      def handle(input)
+        subvolumes_handler if input.to_s == "subvolumes"
```

Comment:
"subvolumes" is a quite weak input name. Very easy to conflict with something else in the dialog. What about `#{id}--#{subvolumes}"?